src/compiler/checker.ts

```diff
@@ -23922,12 +23922,26 @@ namespace ts {
             return getSpellingSuggestion(name, symbols, getCandidateName);
             function getCandidateName(candidate: Symbol) {
                 const candidateName = symbolName(candidate);
-                return !startsWith(candidateName, "\"") && candidate.flags & meaning ? candidateName : undefined;
+                if (startsWith(candidateName, "\"")) {
+                    return undefined;
+                }
+
+                if (candidate.flags & meaning) {
+                    return candidateName;
+                }
+
+                if (candidate.flags & SymbolFlags.Alias) {
+                    const target = resolveAlias(candidate);

```

Comment:
I guess you only want to do this if the symbol’s `links.target !== resolvingSymbol`. I would probably write a `tryResolveAlias` which checks for that and then returns either `resolveAlias(symbol)` or `undefined`.